File: dev/test_data/main/java/expected/implementation_specific/input/snippets/Verification/has_balanced_brackets.java
/**
* Check that the square brackets in the text are balanced.
* @param text the text to be checked
*/
public static boolean hasBalancedBrackets(String text) {
  Objects.requireNonNull(text);

  int depth = 0;
  for (int i = 0; i < text.length(); i++) {
    final char character = text.charAt(i);
    if (character == '[') {
      depth++;
    } else if (character == ']') {
      depth--;
      if (depth < 0) {
        return false;
      }
    }
  }
  return depth == 0;
}

File: dev/test_data/main/cpp/expected/implementation_specific/input/snippets/verification/has_balanced_brackets.cpp
bool HasBalancedBrackets(
  const std::wstring& text
) {
  int depth = 0;

  for (const wchar_t character : text) {
    if (character == L'[') {
      ++depth;
    } else if (character == L']') {
      --depth;
      if (depth < 0) {
        return false;
      }
    }
  }

  return depth == 0;
}
